src/Image.php

```diff
@@ -370,39 +398,39 @@ public function get($name)
      * Set any property on the underlying image.
      *
      * This is handy for fields whose name
-     * does not match PHP's varoiable naming conventions, like `"exif-data"`.
-     * 
-     * @param string $name  The property name. 
-     * @param mixed  $value The value to set for this property.
+     * does not match PHP's variable naming conventions, like `"exif-data"`.
+     *
+     * @param string $name The property name.
+     * @param mixed $value The value to set for this property.
      *
      * @return void
      */
-    public function set($name, $value) 
+    public function set(string $name, $value)
     {
         vips_image_set($this->_image, $name, $value);
     }
 
     /**
      * Call any vips operation.
      *
-     * @param string  $name      The operation name. 
-     * @param Image   $instance  The instance this operation is being 
-     *     invoked from.
-     * @param mixed[] $arguments An array of arguments to pass to the operation.
+     * @param string $name The operation name.
+     * @param Image|null $instance The instance this operation is being invoked from.
+     * @param array $options An array of options to pass to the operation.
+     * @param mixed $arguments,... Arguments to pass to the operation.
      *
-     * @return mixed The result(s) of the operation. 
+     * @return mixed The result(s) of the operation.
      */
-    public static function call($name, $instance, $arguments) 
+    public static function call(string $name, $instance, array $options = [], ...$arguments)
     {
         /*
-        echo "call: ", $name, "\n"; 
+        echo "call: ", $name, "\n";
         echo "instance = ";
         var_dump($instance);
         echo "arguments = ";
         var_dump($arguments);
          */
 
-        $arguments = array_merge([$name, $instance], $arguments);
+        $arguments = array_merge([$name, $instance], $options, $arguments);
```

Comment:
I'm getting confused :( Do we really want to make self::call into a varargs function? 

How about separating options out as a separate argument, so it's

```
call(string $name, $instance, array $arguments, array $options = []) : array
```

Now when we call it from a method, we can do:

```
      return self::call($base . "_const", $this, [$op, $other], $options);
```

And avoid the `array_merge()`, at least some of the time.